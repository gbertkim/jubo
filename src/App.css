* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}
html,
body {
  margin: 0;
  padding: 0;
  height: 100vh;
  width: 100vw;
  font-family: -apple-system, BlinkMacSystemFont, avenir next, avenir, helvetica neue, helvetica, ubuntu, roboto, noto,
    segoe ui, arial, sans-serif;
  position: relative;
  overflow: hidden;
}
#boot {
  background-color: rgb(255, 255, 255);
  position: fixed;
  overflow: hidden;
  width: 100%;
  height: 100%;
  max-width: 100vw;
}
#boot .pageContainer {
  position: absolute;
  width: 100vw;
  height: 100vh;
  will-change: transform;
  max-width: 100vw;
}
#boot .pageContainer .pages{
  touch-action: none;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
  width: 100%;
  height: 100%;
  will-change: transform;
  box-shadow: 0 62.5px 30px -10px rgba(50, 50, 73, 0.3), 0 37.5px 75px -37.5px rgba(0, 0, 0, 0.3);
}
.adminWrapper {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: rgb(248, 248, 248);
}
.adminContainer {
  position: relative;
  width: 50%;
  height: 100%;
  overflow: hidden;
  box-shadow: 0 62.5px 30px -10px rgba(50, 50, 73, 0.3), 0 37.5px 75px -37.5px rgba(0, 0, 0, 0.3);
}
button:hover {
  cursor: pointer;
}
@media screen and (min-width: 800px){
  #boot {
    left: 50%;
    transform: translateX(-50%);
    max-width: 50vw;
  }
  #boot .pageContainer {
    max-width: 50vw;
  }
  #boot .pageContainer .pages{
    box-shadow: 0 1px 1px rgba(0,0,0,0.12), 
    0 2px 2px rgba(0,0,0,0.12), 
    0 4px 4px rgba(0,0,0,0.12), 
    0 8px 8px rgba(0,0,0,0.12),
    0 16px 16px rgba(0,0,0,0.12);
  }
}
@media screen and (max-width: 700px){
  .adminContainer {
    width: 100%;
  }
}

@media screen and (max-width: 600px) {
  body, html {
    font-size: 15px;
  }
}
@media screen and (max-width: 400px) {
  body, html {
    font-size: 14px;
  }
}