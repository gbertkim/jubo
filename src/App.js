import './App.css';
import React, {useEffect} from 'react';
import { Route, BrowserRouter as Router } from 'react-router-dom';
import Login from './components/Login/Login';
import SignUp from './components/SignUp/SignUp';
import Dashboard from './components/Dashboard/Dashboard';
import { AuthProvider } from './AuthContext';
import EventPage from './components/EventPage/EventPage';
import ProgramForm from './components/ProgramForm/ProgramForm';
import AnnouncementPage from './components/AnnouncementPage/AnnouncementPage';
import ContactForm from './components/ContactForm/ContactForm';
import config from './components/config.js';
import Jubo from './components/Jubo/Jubo';
import AdminHeader from './components/AdminHeader/AdminHeader';
import Account from './components/Account/Account';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import Welcome from './components/Welcome/Welcome';
function App() {
  useEffect(() => {
    fetch(`${config.API_ENDPOINT}/accounts`)
    .then(res => {
        if (!res.ok)
            return res.json().then(e => Promise.reject(e))
        return res.json()
    })
    .then(names => {
        console.log(names)
    })
    .catch(e => {
        console.log(e)
    }
    )
  },[])
  return (
    <AuthProvider>
      <div className="App">
        <Router>
          <Route path='/jubo/:jubo' exact>
            <Jubo></Jubo>
          </Route>
          <div className='adminWrapper'>
            <div className='adminContainer'>
              {['/admin/:userid', '/admin/:userid/account', '/admin/:userid/event/:eventid', '/admin/:userid/event/:eventid/program', '/admin/:userid/event/:eventid/contact', '/admin/:userid/event/:eventid/announcements'].map(path => {
                return(
                  <Route key={path} path={path}>
                    <AdminHeader />
                  </Route>
                )
              })}
              <Route path='/' exact>
                <Welcome/>
              </Route>
              <Route path='/login'>
                <Login/>
              </Route>
              <Route path='/signup'>
                <SignUp/>
              </Route>
              <PrivateRoute component={Dashboard} path='/admin/:userid' exact/>
              <PrivateRoute component={Account} path='/admin/:userid/account'/>
              <PrivateRoute component={EventPage} path='/admin/:userid/event/:eventid' exact />
              <PrivateRoute component={ProgramForm} path='/admin/:userid/event/:eventid/program' />
              <PrivateRoute component={AnnouncementPage} path='/admin/:userid/event/:eventid/announcements' exact/>
              <PrivateRoute component={ContactForm} path='/admin/:userid/event/:eventid/contact'/>
            </div>
          </div>
        </Router>
      </div>
    </AuthProvider>
  );
}
export default App;
