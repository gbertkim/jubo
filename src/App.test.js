import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from './App'

describe(`App`, () => {
    it('renders a App by default', () => {
      const wrapper = shallow(<App />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
})