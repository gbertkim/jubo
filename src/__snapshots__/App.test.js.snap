// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a App by default 1`] = `
<AuthProvider>
  <div
    className="App"
  >
    <BrowserRouter>
      <Route
        exact={true}
        path="/jubo/:jubo"
      >
        <Jubo />
      </Route>
      <Route
        key="/admin/:userid"
        path="/admin/:userid"
      >
        <AdminHeader />
      </Route>
      <Route
        key="/admin/:userid/account"
        path="/admin/:userid/account"
      >
        <AdminHeader />
      </Route>
      <Route
        key="/admin/:userid/event/:eventid"
        path="/admin/:userid/event/:eventid"
      >
        <AdminHeader />
      </Route>
      <Route
        key="/admin/:userid/event/:eventid/program"
        path="/admin/:userid/event/:eventid/program"
      >
        <AdminHeader />
      </Route>
      <Route
        key="/admin/:userid/event/:eventid/contact"
        path="/admin/:userid/event/:eventid/contact"
      >
        <AdminHeader />
      </Route>
      <Route
        key="/admin/:userid/event/:eventid/announcements"
        path="/admin/:userid/event/:eventid/announcements"
      >
        <AdminHeader />
      </Route>
      <Route
        exact={true}
        path="/"
      >
        <div
          className="adminWrapper"
        >
          <div
            className="adminContainer"
          >
            <Welcome />
          </div>
        </div>
      </Route>
      <Route
        exact={true}
        path="/login"
      >
        <div
          className="adminWrapper"
        >
          <div
            className="adminContainer"
          >
            <Login />
          </div>
        </div>
      </Route>
      <Route
        exact={true}
        path="/signup"
      >
        <div
          className="adminWrapper"
        >
          <div
            className="adminContainer"
          >
            <SignUp />
          </div>
        </div>
      </Route>
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/admin/:userid"
      />
      <PrivateRoute
        component={[Function]}
        path="/admin/:userid/account"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/admin/:userid/event/:eventid"
      />
      <PrivateRoute
        component={[Function]}
        path="/admin/:userid/event/:eventid/program"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/admin/:userid/event/:eventid/announcements"
      />
      <PrivateRoute
        component={[Function]}
        path="/admin/:userid/event/:eventid/contact"
      />
    </BrowserRouter>
  </div>
</AuthProvider>
`;
